lib/index.js

```diff
@@ -236,16 +267,130 @@ module.exports = class OtpRunner {
   }
 
   /**
-   * Builds a graph using OTP
+   * Recreate certain directories to make sure we're starting fresh
+   */
+  async clearFoldersAndFiles () {
+    try {
+      await fs.remove(this.getTargetFolder())
+      await fs.mkdirp(this.getTargetFolder())
+    } catch (e) {
+      this.log.error(e)
+      await this.fail('Failed to recreate graph router folder')
+    }
+  }
+
+  /**
+   * Returns the appropriate Graph.obj filename that OTP outputs after building
+   * a graph.
+   */
+  getGraphObjName () {
+    return this.isUsingOtp2x() ? 'graph.obj' : 'Graph.obj'
+  }
+
+  /**
+   * Downloads all needed files as specified in the manifest.
+   */
+  async downloadFiles () {
+    const downloadTasks = []
+
+    // add task to download OTP jar
+    downloadTasks.push({
+      dest: this.manifest.jarFile,
+      url: this.manifest.jarUrl
+    })
+
+    // add tasks to download files to the base folder
+    if (
+      this.manifest.baseFolderDownloads &&
+        this.manifest.baseFolderDownloads.length > 0
+    ) {
+      this.manifest.baseFolderDownloads.forEach(url => {
+        const splitUrl = url.split('/')
+        downloadTasks.push({
+          dest: path.join(
+            this.getTargetFolder(),
+            splitUrl[splitUrl.length - 1]
+          ),
+          url
+        })
+      })
+    }
+
+    // add tasks to download various GTFS files and then rename them according
+    // to OTP 2.x file naming requirements
+    if (
+      this.manifest.gtfsDownloads &&
+        this.manifest.gtfsDownloads.length > 0
+    ) {
+      this.manifest.gtfsDownloads.forEach(url => {
+        const splitUrl = url.split('/')
+        downloadTasks.push({
+          dest: path.join(
+            this.getTargetFolder(),
+            `gtfs-${splitUrl[splitUrl.length - 1]}`
```

Comment:
Should we only prepend `gtfs-` if the name doesn't start with `gtfs-`?

Reply:
Logic moved to datatools-server.